qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,118 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Ising model"""
+from typing import cast, Optional
+
+import logging
+
+import numpy as np
+
+from qiskit_nature.operators.second_quantization import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+
+from .lattice_model import LatticeModel
+
+logger = logging.getLogger(__name__)
+
+
+class IsingModel(LatticeModel):
+    """The Ising model."""
+
+    @property
+    def coupling_matrix(self) -> np.ndarray:
+        """Return the coupling matrix."""
+        return self.interaction_matrix()
+
+    @classmethod
+    def uniform_parameters(
+        cls,
+        lattice: Lattice,
+        uniform_interaction: complex,
+        uniform_onsite_potential: complex,
+        onsite_interaction: complex = None,
+    ) -> "IsingModel":
+        """Set a uniform interaction parameter and on-site potential over the input lattice.
+
+        Args:
+            lattice: Lattice on which the model is defined.
+            uniform_interaction: The interaction parameter. (or uniform_interaction)
+            uniform_onsite_potential: The on-site potential. (or uniform_external_field)
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction. If this parameter is specified, it will be ignored.
+
+        Returns:
+            The Ising model with uniform parameters.
+        """
+        if onsite_interaction is not None:
+            logger.warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        return cast(
+            IsingModel,
+            super().uniform_parameters(
+                lattice, uniform_interaction, uniform_onsite_potential, None
+            ),
+        )
+
+    @classmethod
+    def from_parameters(
+        cls, interaction_matrix: np.ndarray, onsite_interaction: complex = None
```

Comment:
I prefer to make it an argument-differ rather than a parameter that will be ignored